DSharpPlus.Commands/Processors/TextCommands/TextExtensions.cs

```diff
@@ -0,0 +1,176 @@
+
+    [GeneratedRegex(@"<@!?(\d+)>", RegexOptions.Compiled)]
+    private static partial Regex userMentionRegex();
+
+    [GeneratedRegex(@"<#(\d+)>", RegexOptions.Compiled)]
+    private static partial Regex channelMentionRegex();
+
+    [UnsafeAccessor(UnsafeAccessorKind.Constructor)]
+    private static extern MessageCreatedEventArgs messageCreateEventArgsConstructor();
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Message")]
+    private static extern void messageCreateEventArgsMessageSetter(MessageCreatedEventArgs messageCreateEventArgs, DiscordMessage message);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_MentionedUsers")]
+    private static extern void messageCreateEventArgsMentionedUsersSetter(MessageCreatedEventArgs messageCreateEventArgs, IReadOnlyList<DiscordUser> mentionedUsers);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_MentionedRoles")]
+    private static extern void messageCreateEventArgsMentionedRolesSetter(MessageCreatedEventArgs messageCreateEventArgs, IReadOnlyList<DiscordRole> mentionedRoles);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_MentionedChannels")]
+    private static extern void messageCreateEventArgsMentionedChannelsSetter(MessageCreatedEventArgs messageCreateEventArgs, IReadOnlyList<DiscordChannel> mentionedChannels);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Constructor)]
+    private static extern DiscordMessage messageConstructor();
+
+    [UnsafeAccessor(UnsafeAccessorKind.Constructor)]
+    private static extern DiscordMessage messageConstructor(DiscordMessage message);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Content")]
+    private static extern void messageContentSetter(DiscordMessage message, string content);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Channel")]
+    private static extern void messageChannelSetter(DiscordMessage message, DiscordChannel channel);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_ChannelId")]
+    private static extern void messageChannelIdSetter(DiscordMessage message, ulong channelId);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_guildId")]
+    private static extern void messageGuildIdSetter(DiscordMessage message, ulong? guildId);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Author")]
+    private static extern void messageAuthorSetter(DiscordMessage message, DiscordUser author);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedUsers")]
+    private static extern ref List<DiscordUser> messageMentionedUsersSetter(DiscordMessage message);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedRoles")]
+    private static extern ref List<DiscordRole> messageMentionedRolesSetter(DiscordMessage message);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedChannels")]
+    private static extern ref List<DiscordChannel> messageMentionedChannelsSetter(DiscordMessage message);

```

Comment:
naming convention; pascal case

Reply:
These are private members, why are we using pascal case

Reply:
methods are always pascal case by convention, private or not

Reply:
Ah, yes. These are methods. Not properties. I have eyes. Sometimes I just can't use them.